examples/opensora_hpcai/scripts/train.py

```diff
@@ -529,18 +543,12 @@ def main(args):
         rank_id=rank_id,
     )
 
-    # FIXME: get_dataset_size() is extremely slow when used with bucket_batch_by_length
     if args.bucket_config is None:
         dataset_size = dataloader.get_dataset_size()
     else:
-        # steps per epoch is not constant in bucket config training
-        # FIXME: It is a highly relaxed estimation to ensure enough steps per epoch to sustain training. \
-        # A more precise estimation or run-time infer is to be implemented.
-        dataset_size = math.ceil(num_src_samples / device_num)
+        # Prevent `GeneratorDataset()` from iterating over the whole dataset.
+        dataset_size = 1  # doesn't affect anything when bucketing is enabled
```

Comment:
It's indeed confusing, but there's nothing we can do about it since `model.train()` will show an incorrect number of steps per epoch anyway. I find fixing it to 1 to be less confusing than the previous behavior.